File: Cpp/1184_Distance_Between_Bus_Stops.cpp
class Solution {
public:
    int distanceBetweenBusStops(vector<int>& distance, int start, int destination) {
        int n = distance.size();
        int clockwise_distance = 0;
        int counterclockwise_distance = 0;
        int start_bus_stop = min(start, destination);
        int destination_bus_stop = max(start, destination);
        for(int i=start_bus_stop; i<destination_bus_stop; ++i){
            clockwise_distance += distance[i];
        }
        
        for(int i=destination_bus_stop; i<start_bus_stop+n; ++i){
            counterclockwise_distance += distance[i%n];
        }
        
        return min(clockwise_distance, counterclockwise_distance);
    }
};

File: Kotlin/1184_Distance_Between_Bus_Stops.kt
class Solution {
    fun distanceBetweenBusStops(distance: IntArray, start: Int, destination: Int): Int {
        val n = distance.size
        val startBusStop = minOf(start, destination)
        val destinationBusStop = maxOf(start, destination)
        var clockwiseDistance = 0
        var counterclockwiseDistance = 0
        
        for(i in startBusStop until destinationBusStop){
            clockwiseDistance += distance[i]
        }
        
        for(i in destinationBusStop until startBusStop + n){
            counterclockwiseDistance += distance[i%n]
        }
        
        return minOf(clockwiseDistance, counterclockwiseDistance)
    }
}